src/project.rs

```diff
@@ -143,10 +148,15 @@ pub fn get_project() -> Result<Project, Error> {
     let mut f = File::open(file)?;
     let mut contents = String::new();
     f.read_to_string(&mut contents)?;
-    let p: Project = match toml_exists() {
+    let mut p: Project = match toml_exists() {
         true => toml::from_str(contents.as_str()).unwrap(),
         false => serde_json::from_str(contents.as_str())?
     };
+    p.template_data = BTreeMap::new();
+    p.template_data.insert("name", p.name.clone());
+    p.template_data.insert("prefix", p.prefix.clone());
+    p.template_data.insert("author", p.author.clone());
+    p.template_data.insert("version", p.version.clone().unwrap());
```

Comment:
Would be nice if we could define those in project file as well, use some project-specific tools for example.
```json
"templates": [
    "my_id": "exec get_my_id.sh"
]
```

This could use same system of execution as custom build steps, which we still have to figure out (especially to support both Windows and Linux we need some kind of separation).

Reply:
This is planned, I'm waiting for Build Steps so I can keep the Windows / Linux separation consistent between templating and build steps, since build steps is going to be more complicated I want to tackle that first.

Reply:
Yup, my thinking as well. :+1: